internal/source/image.go

```diff
@@ -190,3 +181,48 @@ func (i *Image) handleUnexpectedPod(ctx context.Context, pod *corev1.Pod) error
 	_ = i.Client.Delete(ctx, pod)
 	return fmt.Errorf("unexpected pod phase: %v", pod.Status.Phase)
 }
+
+func pendingImagePodResult(pod *corev1.Pod) *Result {
+	var messages []string
+	for _, cStatus := range append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...) {
+		if waiting := cStatus.State.Waiting; waiting != nil {
+			if waiting.Reason == "ErrImagePull" || waiting.Reason == "ImagePullBackOff" {
+				messages = append(messages, waiting.Message)
+			}
+		}
+	}
+	return &Result{State: StatePending, Message: strings.Join(messages, "; ")}
+}
+
+// addSecurityContext is responsible for taking a container and defining the
+// relevant security context values. By having a function do this, we can keep
+// that configuration easily consistent and maintainable.
+func addSecurityContext(pod *corev1.Pod) {
+	// Add security context for overall pod
+	pod.Spec.SecurityContext = &corev1.PodSecurityContext{
+		RunAsNonRoot: pointer.Bool(false),
```

Comment:
Add a TODO comment here to remind us to think about how to make this work under restricted?

Reply:
Good call out, updated with a detail description of what we should be thinking about here.